src/LightGraphsDFG/services/LightGraphsDFG.jl

```diff
@@ -0,0 +1,545 @@
+            end
+        end
+    end
+    return nothing
+end
+
+
+"""
+    $(SIGNATURES)
+Retrieve a deep subgraph copy around a given variable or factor.
+Optionally provide a distance to specify the number of edges should be followed.
+Optionally provide an existing subgraph addToDFG, the extracted nodes will be copied into this graph. By default a new subgraph will be created.
+Note: By default orphaned factors (where the subgraph does not contain all the related variables) are not returned. Set includeOrphanFactors to return the orphans irrespective of whether the subgraph contains all the variables.
+"""
+function getSubgraphAroundNode(dfg::LightGraphsDFG{P}, node::T, distance::Int64=1, includeOrphanFactors::Bool=false, addToDFG::LightGraphsDFG=LightGraphsDFG{P}())::LightGraphsDFG where {P <: AbstractParams, T <: DFGNode}
+    if !haskey(dfg.g.metaindex[:label], node.label)
+        error("Variable/factor with label '$(node.label)' does not exist in the factor graph")
+    end
+
+    # Get a list of all unique neighbors inside 'distance'
+	ns = neighborhood(dfg.g, dfg.g[node.label,:label], distance)
+
+	# Copy the section of graph we want
+	# _copyIntoGraph!(dfg, addToDFG, map(n->get_prop(dfg.g, n, :label), ns), includeOrphanFactors)
+	_copyIntoGraph!(dfg, addToDFG, ns, includeOrphanFactors)
+    return addToDFG
+end
+
+"""
+    $(SIGNATURES)
+Get a deep subgraph copy from the DFG given a list of variables and factors.
+Optionally provide an existing subgraph addToDFG, the extracted nodes will be copied into this graph. By default a new subgraph will be created.
+Note: By default orphaned factors (where the subgraph does not contain all the related variables) are not returned. Set includeOrphanFactors to return the orphans irrespective of whether the subgraph contains all the variables.
+"""
+function getSubgraph(dfg::LightGraphsDFG, variableFactorLabels::Vector{Symbol}, includeOrphanFactors::Bool=false, addToDFG::LightGraphsDFG=LightGraphsDFG())::LightGraphsDFG
+    for label in variableFactorLabels
+        if !haskey(dfg.g.metaindex[:label], label)
+            error("Variable/factor with label '$(label)' does not exist in the factor graph")
+        end
+    end
+
+    _copyIntoGraph!(dfg, addToDFG, variableFactorLabels, includeOrphanFactors)
+    return addToDFG
+end
+
+"""
+    $(SIGNATURES)
+Get an adjacency matrix for the DFG, returned as a Matrix{Union{Nothing, Symbol}}.
+Rows are all factors, columns are all variables, and each cell contains either nothing or the symbol of the relating factor.
+The first row and first column are factor and variable headings respectively.
+"""
+function getAdjacencyMatrix(dfg::LightGraphsDFG)::Matrix{Union{Nothing, Symbol}}
+    varLabels = sort(map(v->v.label, getVariables(dfg)))
+    factLabels = sort(map(f->f.label, getFactors(dfg)))
+    vDict = Dict(varLabels .=> [1:length(varLabels)...].+1)
+
+    adjMat = Matrix{Union{Nothing, Symbol}}(nothing, length(factLabels)+1, length(varLabels)+1)
```

Comment:
One thing we should start doing across the entire code base is avoid type instability -- a major performance hog.  It's okay to use Union for function dispatch, thats still fast, but dynamic types require run time type inference each time.  I think this is happening in various places in the current IIF code anyway, but likelihood is high that we will run a campaign to remove all type instability from the code to improve performance.

cc @GearsAD

Reply:
I'm not sure if we will get type stability at the end of IIF v0.7.x (probably not) or rather do that cleanup during IIF v0.8.x cycle.  v0.8.x cannot be as unstable as IIF v0.7.x is though.  v0.7.x is bit of a hot mess.  As long as IIF 0.6.x and v0.8.x are reliable form version control standpoint.  v0.7.x is a little tricky in places.

Reply:
Should we change to missing? is missing and nothing treated the same? As far as I know missing has little performance impact.

Reply:
I don't know missing that well unfortunately.  The union typedef (not function dispatch) is a bigger story.  This is part of the current DFG design so im scared if LightGraphsDFG is out of step then things are going to get weird.  My first reaction is to just update all of DFG in one go.

Reply:
In this case we need a nullable type. TBD - Let's do DFG in one go?

Reply:
I don't think we need to do anything in this case, julia arrays are fast with Nothing unions. 
Union{Nothing,T} = nullable type.

Reply:
FYI [avoid containers with abstract types](https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-containers-with-abstract-type-parameters-1)